compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -2000,6 +2004,28 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
                   }
                }
 
+            TR::Node *node;
+            ListIterator<TR::Block> blockIt(&blocksInNaturalLoop);
+            TR::Block * derivedInductionVarBlock;
+            for (derivedInductionVarBlock = blockIt.getCurrent(); derivedInductionVarBlock; derivedInductionVarBlock = blockIt.getNext())
+               {
+               for (TR::TreeTopIterator treeTopIt(derivedInductionVarBlock->getEntry(),comp()); treeTopIt != derivedInductionVarBlock->getExit(); treeTopIt.stepForward())
+                    {
+                    node = treeTopIt.currentNode();
+                    if (node->getSymbol() == derivedInductionVar->getSymbol())
+                        goto trivialArrayOperationCheck;
+                    }
+               }
+
+            trivialArrayOperationCheck:
+            TR_ASSERT_FATAL(node->getSymbol() == derivedInductionVar->getSymbol(),"Node symbol is not the same as derivedInductionVar");
+            TR::NodeChecklist visited(comp());
+            if (!disableTrivialArrayOperationCheck && isTrivialArrayOperation(visited, node))
+               {
+               dumpOptDetails(comp(), "Skipping induction variable elimination for %d as it is being used in a trivial array operation.\n", derivedInductionVar->getReferenceNumber());
+               goto moveToNextIndVar;
```

Comment:
This should be guarded via `performTransformation` as it is a transformation after all (i.e. we are choosing not to do something we otherwise would).